### PR TITLE
Fix node programgen enum to use package identifier.

### LINKED ISSUE #12986: TS programgen bug regarding enums
### What happened?

I did a `pulumi import` of a [webhook](https://www.pulumi.com/registry/packages/pulumiservice/api-docs/webhook/) and it produced the following code: 

```typescript
const datalake = new pulumiservice.Webhook("datalake", {
    active: true,
    displayName: [elided],
    format: @pulumi/pulumiservice.WebhookFormat.Slack,
    organizationName: "pulumi",
    payloadUrl: [elided],
    projectName: "datalake",
    stackName: "datawarehouse-production",
}, {
    protect: true,
});
```

It says `@pulumi/pulumiservice.WebhookFormat.Slack` but it should just be `pulumiservice.WebhookFormat.Slack`

### Expected Behavior

Expected `@pulumi/pulumiservice.WebhookFormat.Slack` to be `pulumiservice.WebhookFormat.Slack`

### Steps to reproduce

create a webhook in the pulumi UI and import it using the pulumiservice provider

### Output of `pulumi about`

CLI          
Version      3.62.0
Go Version   go1.20.2
Go Compiler  gc

Plugins
NAME           VERSION
nodejs         unknown
pulumiservice  0.7.1

Host     
OS       darwin
Version  13.0.1
Arch     arm64

This project is written in nodejs: executable='/Users/komal/.nvm/versions/node/v18.12.0/bin/node' version='v18.12.0'

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/komal-pulumi-corp
User           komal-pulumi-corp
Organizations  komal-pulumi-corp, service-provider-test-org, pulumi

Dependencies:
NAME                   VERSION
@pulumi/pulumi         3.62.0
@pulumi/pulumiservice  0.7.1
@types/node            16.18.23

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by dixler**
# Description

Fixes #12986 

Fixes issue with javascript program generation where enums would incorrectly reference the package rather than the import alias.

There are changes being made to codegen testing (adding matrix testing) which is why a more extensive test is not being added.

Context:

NodeJS package aliases are generated using the following code
https://github.com/pulumi/pulumi/blob/917b2afdfcb9d797f5d756f6e9084717d2c57da6/pkg/codegen/nodejs/gen_program.go#L333

When a provider schema supplies a language specific package name for nodejs like

```json
{
  "name": "pulumiservice",
  "language": {
    ...
    "nodejs": {
      "packageName": "@pulumi/bar",
      ...
    },
    ...
  },
  ...
}
```

programgen would use `@pulumi/bar` rather than `pulumiservice`. for enums

```typescript
import * as pulumi from "@pulumi/pulumi";
import * as pulumiservice from "@pulumi/bar";

const my_webhook = new pulumiservice.Webhook("my_webhook", {
    active: true,
    displayName: "kyle-repro",
    filters: [@pulumi/bar.WebhookFilters.DeploymentSucceeded],
    format: @pulumi/bar.WebhookFormat.Raw,
    organizationName: "pulumi",
    payloadUrl: "https://example.com",
    projectName: "awstypescript",
    stackName: "dev",
}, {
    protect: true,
});
```

this PR fixes it to correctly show as follows:

```typescript
import * as pulumi from "@pulumi/pulumi";
import * as pulumiservice from "@pulumi/bar";

const my_webhook = new pulumiservice.Webhook("my_webhook", {
    active: true,
    displayName: "kyle-repro",
    filters: [pulumiservice.WebhookFilters.DeploymentSucceeded],
    format: pulumiservice.WebhookFormat.Raw,
    organizationName: "pulumi",
    payloadUrl: "https://example.com",
    projectName: "awstypescript",
    stackName: "dev",
}, {
    protect: true,
});
```

## Checklist

- [x] I have run `make tidy` to update any new dependencies
- [x] I have run `make lint` to verify my code passes the lint check
  - [x] I have formatted my code using `gofumpt`

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Cloud API version

### CODE REVIEW

File: pkg/codegen/nodejs/gen_program_expressions.go [hunk 1]
```diff
@@ -309,13 +309,6 @@ func enumName(enum *model.EnumType) (string, error) {
 	if !ok {
 		return "", fmt.Errorf("Could not get associated enum")
 	}
-	def, err := e.(*schema.EnumType).PackageReference.Definition()
-	if err != nil {
-		return "", err
-	}
-	if name := def.Language["nodejs"].(NodePackageInfo).PackageName; name != "" {
-		pkg = name
-	}
```
Comment (dixler) [inline pkg/codegen/nodejs/gen_program_expressions.go:318]: @iwahbe is there anything I need to be worried about if I delete these lines?

tl;dr working on a bug in nodejs programgen for enums

Current state:

```typescript
import * as pulumi from "@pulumi/pulumi";
import * as pulumiservice from "@pulumi/pulumiservice";

const my_webhook = new pulumiservice.Webhook("my_webhook", {
    active: true,
    displayName: "kyle-repro",
    filters: [@pulumi/pulumiservice.WebhookFilters.DeploymentSucceeded],
    format: @pulumi/pulumiservice.WebhookFormat.Raw,
    organizationName: "pulumi",
    payloadUrl: "https://example.com",
    projectName: "awstypescript",
    stackName: "dev",
}, {
    protect: true,
});
```

PR state:

```typescript
import * as pulumi from "@pulumi/pulumi";
import * as pulumiservice from "@pulumi/pulumiservice";

const my_webhook = new pulumiservice.Webhook("my_webhook", {
    active: true,
    displayName: "kyle-repro",
    filters: [pulumiservice.WebhookFilters.DeploymentSucceeded],
    format: pulumiservice.WebhookFormat.Raw,
    organizationName: "pulumi",
    payloadUrl: "https://example.com",
    projectName: "awstypescript",
    stackName: "dev",
}, {
    protect: true,
});
```

Comment (iwahbe) [reply pkg/codegen/nodejs/gen_program_expressions.go:318]: If `pulumiservice` has overriden the nodejs package name to be `@pulumi/bar`, you should generate `bar.WebhookFilters.DeploymentSucceeded` instead of `pulumiservice.WebhookFilters.DeploymentSucceeded`. Those lines accommodate for setting `PackageName` in the enums host schema.

Comment (dixler) [reply pkg/codegen/nodejs/gen_program_expressions.go:318]: had internal discussion around test capturing expectations if a different package alias identifier is used.

**Review by iwahbe — APPROVED** [review]

**Review by Zaid-Ajaj — APPROVED** [review]
LGTM

**Comment by dixler** [comment]
bors merge
